src/services/fileStorage/utils/copyCourseFiles.js

```diff
@@ -0,0 +1,65 @@
+const { NotFound, BadRequest } = require('../../../errors');
+const { FileModel } = require('../model');
+
+const { equal: equalIds } = require('../../../helper/compare').ObjectId;
+const { generateFileNameSuffix } = require('./filePathHelper');
+const { createDefaultPermissions } = require('./createDefaultPermissions');
+const { canRead } = require('./filePermissionHelper');
+
+const safeOverrideAndClear = (source, override) => {
+	Object.keys(override).forEach((key) => {
+		source[key] = override[key];
+	});
+	delete source._id;
```

Comment:
whitelist with mapping / mapper class is better i think. Legacy code or not.

Reply:
as discussed: change is too big for now